examples/branch-protection/main.tf

```diff
@@ -0,0 +1,51 @@
+data "github_team" "admin" {
+  slug = "AdminTestTeam"
+}
+
+data "github_team" "maintain" {
+  slug = "MaintainTestTeam"
+}
+
+locals {
+  admin    = data.github_team.admin.id
+  maintain = data.github_team.maintain.id
+}
+
+# ###############################################################################
+# The examples below create an organization repository and team repository
+# To deploy this example export both the GITHUB_TOKEN and GITHUB_OWNER variables
+# ###############################################################################
+
+module "branch_protection" {
+  source                   = "./../../"
+  name                     = "sample-terraform-repository"

```

Comment:
Names of the repos must be unique on the examples, we are going to create a makefile which will create all examples in sequence, suggestions:
`example-branch-protection`
`example-minimum`
`example-complete`